### PR TITLE
Use MathJax format for complexity annotations

### LINKED ISSUE #731: All complexity annotations should be in MathJax format
…because it's beautiful!

MathJax is already used in:

* `Data.Sequence`
* `compose` (as of https://github.com/haskell/containers/pull/729)

### PR DESCRIPTION
**Pull request by konsumlamm**
Closes #731.

### CODE REVIEW

File: containers/src/Data/Map/Internal.hs [hunk 1]
```diff
@@ -1800,7 +1800,7 @@ unionsWith f ts
 {-# INLINABLE unionsWith #-}
 #endif
 
--- | /O(m*log(n\/m + 1)), m <= n/.
+-- | \(O(m \log(n/m+1)), m \leq n\).
```
Comment (treeowl) [inline containers/src/Data/Map/Internal.hs:1803]: I think we can take advantage of the better layout and so on MathJax offers us to fix the nagging partiality in this expression.

```suggestion
-- | \($O\bigl(m \log \bigl(\frac{n+1}{m+1}\bigr)\bigr), \quad m \leq n$\).
```

If you think a `\quad` space is too big, you can use a `\;` one. I'm not sure which is better.

File: containers/src/Data/Map/Internal.hs [hunk 2]
```diff
@@ -1824,7 +1824,7 @@ union t1@(Bin _ k1 x1 l1 r1) t2 = case split k1 t2 of
 {--------------------------------------------------------------------
   Union with a combining function
 --------------------------------------------------------------------}
--- | /O(m*log(n\/m + 1)), m <= n/. Union with a combining function.
+-- | \(O(m \log(n/m+1)), m \leq n\). Union with a combining function.
```
Comment (treeowl) [inline containers/src/Data/Map/Internal.hs:1827]: Same as above for all these.

File: containers/src/Data/Sequence/Internal.hs [hunk 3]
```diff
@@ -792,7 +792,7 @@ squashR (One12 n) m = node2 n m
 squashR (Two12 n1 n2) m = node3 n1 n2 m
 
 
--- | /O(m*n)/ (incremental) Takes an /O(m)/ function and a finger tree of size
+-- | \(O(m*n)\) (incremental) Takes an \(O(m)\) function and a finger tree of size
 -- /n/ and maps the function over the tree leaves. Unlike the usual 'fmap', the
```
Comment (treeowl) [inline containers/src/Data/Sequence/Internal.hs:796]:
```suggestion
-- | \(O(mn)\) (incremental) Takes an \(O(m)\) function and a finger tree of size
-- \(n\) and maps the function over the tree leaves. Unlike the usual 'fmap', the
```

File: containers/src/Data/Map/Internal.hs (hunk 1 above)
Comment (konsumlamm) [reply containers/src/Data/Map/Internal.hs]: I changed it to `\(O(m \log\bigl(\frac{n+1}{m+1}\bigr)), \; m \leq n\)`, I hope that's okay? It keeps the O(...) consistent with the other annotations.

It looks like this:
example

**Comment by konsumlamm** [comment]
> So much work!

It isn't that much work, just a few search & replace operations. Just someone had to do it.

File: containers/src/Data/Map/Internal.hs (hunk 1 above)
Comment (treeowl) [reply containers/src/Data/Map/Internal.hs]: No, small parentheses around bigger ones look wrong. What's inconsistent?

Comment (konsumlamm) [reply containers/src/Data/Map/Internal.hs]: The inconsistency is that other O(...) annotations use normal parantheses instead of the big ones.

Comment (treeowl) [reply containers/src/Data/Map/Internal.hs]: That kind of inconsistency is just fine. Standard mathematical practice.

Comment (konsumlamm) [reply containers/src/Data/Map/Internal.hs]: Ok, I'll add the big parentheses.

**Comment by konsumlamm** [comment]
@treeowl is there anything left to do before this can be merged?

**Comment by treeowl** [comment]
No, I just forgot.

**Comment by treeowl** [comment]
Thanks again.

**Pull request merged**